detect/git.go

```diff
@@ -28,6 +29,9 @@ func (d *Detector) DetectGit(cmd *sources.GitCmd, remote *RemoteInfo) ([]report.
 	// loop to range over both DiffFiles (stdout) and ErrCh (stderr)
 	for diffFilesCh != nil || errCh != nil {
 		select {
+		case <-ctx.Done():
+			// Return context error if context is cancelled
+			return d.findings, ctx.Err()
```

Comment:
This is awkward. IMO we'd either need to defer `sema.wait()` or call `break` here.